File: samples/minimax-helper-python/bot.py
"""Sample bot: minimax (depth 4) with a heuristic. Thin wrapper: the real
search lives in search.py."""

from search import minimax_move


def choose_move(board, you, info=None):
    return minimax_move(board, you)


def legal_moves(board):
    return [col for col, column in enumerate(board) if column[-1] == 0]

File: samples/minimax-rust/src/bot.rs
//! Sample bot: minimax (depth 4), same algorithm as the TypeScript sample.
//! Thin wrapper: the real search lives in search.rs.
mod search;

pub struct MoveInfo {
    pub moves: Vec<usize>,
    pub match_id: String,
    pub game_number: u32,
    pub clock_remaining_ms: u64,
}

pub fn choose_move(board: &[Vec<u8>], you: u8, _info: &MoveInfo) -> usize {
    search::minimax_move(board, you)
}

pub fn legal_moves(board: &[Vec<u8>]) -> Vec<usize> {
    search::legal_moves(board)
}
